### PR TITLE
Implement Token Grant KYC Handle

### LINKED ISSUE #5905: TokenGrantKycToAccountHandler
Implement `TokenGrantKycToAccountHandler.handle(...)`

Also creates `WritableTokenRelationStore`

### PR DESCRIPTION
**Pull request by mhess-swl**
Closes #5905

### CODE REVIEW

**Comment by mhess-swl** [comment]
@Neeharika-Sompalli there are a couple of javadoc compilation failures on the `TokenUn/PauseHandler` classes, but I wouldn't think those are related to this work. If you can review this code sometime next week, then I can work on it when I get back :)

**Review by Neeharika-Sompalli — COMMENTED** [review]
Need to merge develop back into this PR  🙂 @mhess-swl

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGrantKycToAccountHandler.java [hunk 1]
```diff
@@ -70,7 +72,22 @@ public void preHandle(@NonNull final PreHandleContext context, @NonNull final Re
...
+        final var targetTokenId = op.tokenOrThrow();
+        final var targetAccountId = op.accountOrThrow();
+        final var tokenRelation =
+                tokenRelStore.getForModify(targetTokenId.tokenNum(), targetAccountId.accountNumOrThrow());
+
+        /* --- Persist the updated models --- */
+        final var tokenRelBuilder = tokenRelation.orElseThrow().copyBuilder();
+        tokenRelBuilder.kycGranted(true);
+        tokenRelStore.put(tokenRelBuilder.build());
+        tokenRelStore.commit();
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGrantKycToAccountHandler.java:91]: We should probably move this to `TransactionDispatcher` dispatch method. To keep all `commit` in one place.

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableTokenRelationStore.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                .put(
+                        EntityNumPair.fromLongs(tokenRelation.tokenNumber(), tokenRelation.accountNumber()),
+                        Objects.requireNonNull(tokenRelation));
+    }
+
+    /**
+     * Commits the changes to the underlying data storage
+     */
+    public void commit() {
+        requireNonNull(tokenRelState);
```
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableTokenRelationStore.java:67]: I'd like to get this PR merged to build on top of it. Ok if I include this change in my next PR?

**Review by edward-swirldslabs — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**